cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
```

Comment:
Yes, it does something similar for MSVC, It becomes a /D "BOOST_ALL_NO_LIB=1 BOOST_ALL_DYN_LINK", however this does work for the MSVC compiler and considering this will only be used for MSVC it will work, but definitely a hack.